### PR TITLE
#107 solution to blockers linking task has been implemented

### PR DESCRIPTION
**Pull request by yeain-shukla**
#107 solution to blockers linking task has been implemented

### CODE REVIEW

File: src/main/java/com/groupesan/project/java/scrumsimulator/mainpackage/ui/panels/SolutionListPane.java [hunk 1]
```diff
@@ -20,24 +20,24 @@
...
         setSize(500, 300);
 
-        GridBagLayout myGridbagLayout = new GridBagLayout();
-        JPanel myJpanel = new JPanel();
-        myJpanel.setBorder(new EmptyBorder(10, 10, 10, 10));
-        myJpanel.setLayout(myGridbagLayout);
+        GridBagLayout layout = new GridBagLayout();
+        JPanel mainPanel = new JPanel();
+        mainPanel.setBorder(new EmptyBorder(10, 10, 10, 10));
+        mainPanel.setLayout(layout);
```
Comment (thrupthihm) [inline src/main/java/com/groupesan/project/java/scrumsimulator/mainpackage/ui/panels/SolutionListPane.java:40]: @yeain-shukla Can you please elaborate, why they are changes in this file SolutionListPane.java, which are not necessary to make changes related to your task, I see just the variable names have been changed and few lines have unnecessary updates, please retain only changes needed for task and update the PR accordingly.
Thank you.

File: src/main/java/com/groupesan/project/java/scrumsimulator/mainpackage/ui/panels/SolutionListPane.java [hunk 2]
```diff
@@ -46,51 +46,47 @@ public void init(){
...
-        myJpanel.add(
+        mainPanel.add(
                 scrollPane,
                 new CustomConstraints(0, 0, GridBagConstraints.WEST, 1.0, 0.8, GridBagConstraints.BOTH)
         );
 
-        JButton newSolutionButton = new JButton("Add solution");
-        newSolutionButton.addActionListener(
-                new ActionListener() {
+        JButton newSolutionButton = new JButton("Add Solution");
```
Comment (thrupthihm) [inline src/main/java/com/groupesan/project/java/scrumsimulator/mainpackage/ui/panels/SolutionListPane.java:54]: same here as well

File: src/main/java/com/groupesan/project/java/scrumsimulator/mainpackage/ui/widgets/BlockerWidget.java [hunk 3]
```diff
@@ -129,15 +139,24 @@ private void linkUserStoryToBlocker() {
...
+                null,
+                solutions.toArray(),
+                null);
+
+        if (selectedSolution != null) {
+            blocker.resolve(selectedSolution);
+
+            solutionLabel.setText("Solution: " + selectedSolution.getTitle());
+            resolveButton.setText("Resolved");
+            resolveButton.setEnabled(false);
```
Comment (thrupthihm) [inline src/main/java/com/groupesan/project/java/scrumsimulator/mainpackage/ui/widgets/BlockerWidget.java:157]: Simultaneously after this step, please store the solutionId (Chosen for resolving blocker) and blockerId( Resolved with solution), add under respective blocker and solution in simulation.json as we discussed during design.

**Comment by thrupthihm** [comment]
Also please resolve conflicts and keep both the changes while resolving.

File: src/main/java/com/groupesan/project/java/scrumsimulator/mainpackage/ui/panels/SolutionListPane.java (hunk 2 above)
Comment (yeain-shukla) [reply src/main/java/com/groupesan/project/java/scrumsimulator/mainpackage/ui/panels/SolutionListPane.java:54]: there is no difference i just kept new ActionListener() in the same line as newSolutionButton.addActionListener. there is no difference in functionality

File: src/main/java/com/groupesan/project/java/scrumsimulator/mainpackage/ui/widgets/BlockerWidget.java (hunk 3 above)
Comment (yeain-shukla) [reply src/main/java/com/groupesan/project/java/scrumsimulator/mainpackage/ui/widgets/BlockerWidget.java:157]: OKay, I'll create a new issue and start working on this functionality.

**Pull request closed without merging**